File: tiq-e/strings/5/main.go
package main

import (
	"fmt"
	"regexp"
	"strings"
	"time"
)

func isPalindrome(s string) bool {
	var reg, _ = regexp.Compile("[^a-z0-9]+")
	s = reg.ReplaceAllString(strings.ToLower(s), "")

	c1, c2 := 0, len(s)-1

	for c1 < c2 {
		if s[c1] != s[c2] {
			return false
		}
		c1++
		c2--
	}
	return true
}

func main() {
	s := "aba"

	var start = time.Now()
	var result = isPalindrome(s)
	var end = time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}

File: tiq-e/strings/5/main.ts
import {performance} from 'perf_hooks';

function isPalindrome(s: string): boolean {
    // let string = s.replace(/[^a-z0-9]/ig, "").toLowerCase();
    // return string === string.split("").reverse().join("");

    let a1 = s
        .replace(/[^a-z0-9]/ig, "")
        .toLowerCase()
        .split('')

    let a2 = a1.reverse()
    let a3 = a1.map((e, i) => [e, a2[i]])

    return a3.reduce((acc: true | false, [first, second]) => {
        acc = first === second;
        return acc
    }, true)
}


const s = "ab_as";

const start = performance.now();
const result = isPalindrome(s);
const end = performance.now();
console.log('Duration:', end - start);
console.log('Result:', result)
